src/block-management/block-manager.js

```diff
@@ -178,8 +207,9 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 				<TextInput
 					multiline
 					numberOfLines={ 0 }
-					style={ styles.htmlView }>
-					{ this.serializeToHtml() }
+					style={ styles.htmlView }
+					onChangeText={ ( html ) => this.props.html = html }>
```

Comment:
You should never mutate `props`, you should track changes to `html` in `state`. In this case, you rather should pass down callback to this component which will update `html` outside.

Reply:
My original idea was to keep track of the `html` updates in a simple property `this.html`, but I kept getting a `flow` warning about it not being part of the `BlockManager` type.

I tried to add it as part of the component `state` but on every update, the InputView is reloaded and it feels weird from a UX point of view (but it works).

What would be the advantage of lifting the `html` state up vs keeping it in the component's state?

Reply:
I also tried what is described [here (lifting state up)](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up), but the jumpiness remains.